neuralprophet/forecaster.py

```diff
@@ -726,18 +792,30 @@ def fit(self, df, freq="auto", validation_df=None, progress="bar", minimal=False
         df = self._check_dataframe(df, check_y=True, exogenous=True)
         self.data_freq = df_utils.infer_frequency(df, n_lags=self.max_lags, freq=freq)
         df = self._handle_missing_data(df, freq=self.data_freq)
-        if validation_df is not None and (self.metrics is None or minimal):
-            log.warning("Ignoring validation_df because no metrics set or minimal training set.")
-            validation_df = None
 
         # Training
         if validation_df is None:
-            metrics_df = self._train(df, minimal=minimal, continue_training=continue_training)
+            metrics_df = self._train(
+                df,
+                progress=bool(progress),
+                metrics=bool(self.metrics),
+                checkpointing=checkpointing,
+                continue_training=continue_training,
+                num_workers=num_workers,
+            )
         else:
             df_val, _, _, _ = df_utils.prep_or_copy_df(validation_df)
             df_val = self._check_dataframe(df_val, check_y=False, exogenous=False)
             df_val = self._handle_missing_data(df_val, freq=self.data_freq)
-            metrics_df = self._train(df, df_val=df_val, minimal=minimal, continue_training=continue_training)
+            metrics_df = self._train(
+                df,
+                df_val=df_val,
+                progress=bool(progress),
```

Comment:
Same as with metrics comment above, we should have clear variable naming if they contain different values in different places